extendeddatacrossword.go

```diff
@@ -473,7 +466,14 @@ func (eds *ExtendedDataSquare) computeSharesRootWithRebuiltShare(shares [][]byte
 
 // verifyEncoding checks the Reed-Solomon encoding of the provided data.
 func (eds *ExtendedDataSquare) verifyEncoding(data [][]byte, rebuiltIndex int, rebuiltShare []byte) error {
-	data[rebuiltIndex] = rebuiltShare
+	if rebuiltShare != nil && rebuiltIndex >= 0 {
+		data[rebuiltIndex] = rebuiltShare
+		defer func() {
```

Comment:
Style: do we want to use `defer` here? AFAIK, no where else in rsmt2d uses `defer`, primarily because it's a Go-specific thing that doesn't translate well into a language like C (C being a lower common denominator for readability). I don't have strong opinions, so opening for discussion.

Reply:
I'm in favor of using defer because it ensures cleanup operations are closely tied to their corresponding setups. This not only makes the code more robust by preventing omission in multiple return paths but also aligns with good programming practices by keeping related actions together, improving both maintainability and readability.

An alternative would be to manually add a revert of the addition to the data slice in every return statement, which could be error-prone. 

Another option is to adopt an immutable approach by creating a copy of the data that includes the extra share, ensuring the original data remains unmodified. However, this method would lead to extra allocations, which is why I haven't used it here.

Reply:
I see.

Hmmmm. Perhaps creating a copy wouldn't be so bad if it's only for a single row or column. Could you open an issue that someone can work on to investigate the actual performance impact?

Other that than, no further comments. Not blocking merging.

Reply:
Defer translates well into C or any other language by calling the deferred line after everything else, so we don't gain much by going with the alternative approach and only lose time (investigating performance impact through the issue) and readability(no defer based grouping). Overall, I don't think opening and spending time on the issue is worth it

Reply:
Ok. I'll defer (get it 😉) to you guys on this.